localstack-core/localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -770,6 +771,8 @@ def _replace(match):
 
                 if parameter_type in ["CommaDelimitedList"] or parameter_type.startswith("List<"):
                     return [p.strip() for p in parameter_value.split(",")]
+                elif parameter_type == "Number":
+                    return str(parameter_value)
```

Comment:
This converts the number to string first so that `re.sub(regex, _replace, result)` is replaced by the correct values without any type errors.

Reply:
This _seems_ a little hacky, but I think the implementation here is fine. As you say the contract for `re.sub` requires the `repl` argument (if callable) to return a `str` 👍